src/core/qgsproject.cpp

```diff
@@ -975,6 +983,17 @@ void QgsProject::onMapLayersAdded( const QList<QgsMapLayer*>& layers )
     }
 
     connect( layer, SIGNAL( configChanged() ), this, SLOT( setDirty() ) );
+
+    // check if we have to update connections for layers with dependencies
+    for ( QMap<QString, QgsMapLayer*>::iterator it = existingMaps.begin(); it != existingMaps.end(); it++ )
+    {
+      QSet<QString> deps = it.value()->dataDependencies();
+      if ( deps.contains( layer->id() ) )
+      {
+        // reconnect to change signals
+        it.value()->setDataDependencies( deps );
```

Comment:
No the dependency is still there, but no connection is made if the other layer is not there yet.

This is to cover the case where you have a layer A that depends on B and you load them one after the other (embbeded layers for instance)
* A is loaded, B is not there -> no signal connection
* B is loaded afterward and it will reset A's connections, B is there now and the connection can be made

And ... I even realize this has to be done recursively (if A -> B -> C and they are loaded in that order)